### PR TITLE
Fix some wording, add examples

### PR DESCRIPTION
**Pull request by black-sliver**

### CODE REVIEW

File: Protocol-Draft-v3.md [hunk 1]
```diff
@@ -86,26 +88,31 @@ The first byte of the reply indicates its type : `\n` for an ascii reply or `\0`
...
-A key name is lower case and separated with _. Value format is free since its depends on its meanings.
-
-For predefined values (like emulator state) these values are in lower case.
+Key names are lower case and separated by _. Values are arbitrary strings and their meaning depends on the command.
 
-Keys can repeat, the data then represents a `list<map<string,string>>`.
+For predefined constants (like emulator state) values are in lower case.
 
-Whenever a duplicate key is received, a new map is started in the list.
+Keys can repeat, the data then represents a `list<map<string,string>>`:
```
Comment (Skarsnik) [inline Protocol-Draft-v3.md]: Don't use C++ Template notation, just says it's an array of hash

File: Protocol-Draft-v3.md [hunk 2]
```diff
@@ -86,26 +88,31 @@ The first byte of the reply indicates its type : `\n` for an ascii reply or `\0`
...
 
-**TODO** : Clarify this.
+Whenever a duplicate key is received, a new map is started in the list, so to generate the list
```
Comment (Skarsnik) [inline Protocol-Draft-v3.md]: Maybe an example, implementation details are not meant to be here.

File: Protocol-Draft-v3.md [hunk 3]
```diff
@@ -265,10 +272,21 @@ Offsets/addresses that start with $ are hexadecimal, otherwise they are decimal.
 
 * If offset is empty, read entire memory.
 * If size is empty, read from offset to end.
+* If offsetN is given, sizeN can not be omitted for N>=2
 * If reading out of bounds for last offset/size: reply can be short (reply has less bytes than requested).
 * If reading out of bounds for non-last offset/size: reply has to be padded with 0.
 * If **all** addresses are out of bounds the reply can be empty instead of sending padding.
-* If offsetN is given, sizeN can not be omitted for N>=2
+
```
Comment (Skarsnik) [inline Protocol-Draft-v3.md:288]: Again implementation details x)

Comment (black-sliver) [reply Protocol-Draft-v3.md:288]: ~~this is not an implementation detail. the command is invalid if the length is missing for reads/writes that have multiple regions~~ oh, you were talking about the lines that follow.

I've been asked specifically to explain *how* to correctly interpret and return the regions

**Pull request merged**